curvefs/src/metaserver/metaserver.cpp

```diff
@@ -87,6 +89,8 @@ void Metaserver::InitRegisterOptions() {
     conf_->GetValueFatalIfFail("global.external_ip",
                                &registerOptions_.metaserverExternalIp);
     conf_->GetValueFatalIfFail("global.port", &registerOptions_.metaserverPort);
+    conf_->GetValueFatalIfFail("global.external_port",
```

Comment:
conf->GetBoolValue("global.enable_external_server",
        &options_.enableExternalServer);

if enable is false, can you still get external port?

Reply:
> ```
> conf->GetBoolValue("global.enable_external_server",
>     &options_.enableExternalServer);
> ```
> 
> if enable is false, can you still get external port?

Default the switch is off and external ip and port is same as internal ip and port. The metaserve.conf and topology.json created by curveamd will support and I have reached a consensus with @Wine93.
If you want to turn on this switch, you need set enable to true and available external ip and port in topology.yaml(curveamd).